src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+/**
+ * Feel free to remove this class and create your own.
+ */
+public class Main {
+    private static final FigureSupplier figureSupplier = new FigureSupplier();
+
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
+        for (int i = 0; i < figures.length; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+            if (i > (figures.length - 1) / 2) {
+                figures[i] = figureSupplier.getDefaultFigure();
+            }
```

Comment:
The condition in the 'if' statement is incorrect. It will always evaluate to true for the last half of the array elements, which is not necessary. You should create all random figures before the loop and then use the loop only to call the draw() method.